core/drivers/crypto/crypto_api/acipher/ecc.c

```diff
@@ -106,7 +129,7 @@ static TEE_Result ecc_generate_keypair(struct ecc_keypair *key,
 		return TEE_ERROR_BAD_PARAMETERS;
 	}
 
-	key_size_bits = get_ecc_key_size_bytes(key->curve) * 8;
+	key_size_bits = get_ecc_key_size_bits(key->curve);
```

Comment:
Will this work with the CAAM driver? https://github.com/OP-TEE/optee_os/blob/3.18.0/core/drivers/crypto/caam/acipher/caam_ecc.c#L227 explicitly mention "multiple of 8 bits" @cneveux @clementfaure

Reply:
For `TEE_ECC_CURVE_NIST_P521`, the CAAM needs to allocate 528 bits (66 bytes). 
I would add 
```
static TEE_Result do_gen_keypair(struct ecc_keypair *key, size_t key_size)
...
	ECC_TRACE("Generate Keypair of %zu bits", key_size);

	/* The key size must be a multiple of 8 bits */
	key_size = ROUNDUP(key_size, 8);
...
```
in `caam_ecc.c`.